### ISSUE TITLE
ModPagespeedMapOriginDomain doesn't work

### ISSUE THREAD

**Issue by rcpires** [apache/incubator-pagespeed-mod#1243]
Hi, I have mapped my domain and subdomains with ModPagespeedMapOriginDomain:

 ModPagespeedDomain *.domain.com
ModPagespeedMapOriginDomain http://domain.com https://domain.com
ModPagespeedMapOriginDomain http://sub.domain.com https://sub.domain.com

But I got a lot of error messages on log like that:

Rewrite https://sub.domain.com/dados/tema9_img_usuario_t1Tji_1427153187.jpg failed while fetching https://sub.domain.com/dados/xtema9_img_usuario_t1Tji_1427153187.jpg.pagespeed.ic.dkD6hTULmH.jpg

So pagespeed fetcher is trying get resource using https (ignore mapping) but fail because I use /etc/hosts to resolve my domains to 127.0.0.1 (and my ssl terminator is on load balancer). I can't use ModPagespeedMapOriginDomain http://localhost https://domain.com because there are a lot of logic on application based on what subdomain is being accessed.

What can be wrong pls?

**Comment by oschaaf** [comment]
Perhaps you could use `MapOriginDomain` to point pagespeed to the load balancer/ssl terminator?

**Comment by oschaaf** [comment]
Sorry, that went out too early. Perhaps you could create another entry in /etc/hosts that does point to the load balancer/ssl terminator just for pagespeed, and set that new hostname up to work (forward it correctly to the webserver, make the webserver listen to that host)?

**Comment by rcpires** [comment]
Yeah, this its an option, but forward pagespeed to load balancer no makes sense, MapOriginDomain exists just for avoid that. I could forward to own server (localhost or another hostname and entry in /etc/hosts) and try modify app to treat the subdomain logic. But my doubt is why my actual mappping not work? Thanks!
